### ISSUE TITLE
Are you interested in doing an onnx export for other work?  

### ISSUE THREAD

**Issue by GavinYang5** [fabio-sim/DeDoDe-ONNX-TensorRT#7]
Thank you so much for the great work you do
I'm currently trying to export this work to the onnx model https://github.com/qinzheng93/GeoTransformer
I'm having trouble with the conversion
It's been a long time since the original author maintained it, I know you've done a great job with Torch2Onnx
If you can help me, I would appreciate it.

**Comment by fabio-sim** [comment]
Hi @GhYang0519, thank you for your interest.

I'd be glad to help. What problems are you facing when exporting GeoTransformer to ONNX?

**Comment by GavinYang5** [comment]
I'm so glad you got back to me
![image](https://github.com/fabio-sim/DeDoDe-ONNX-TensorRT/assets/93421835/cfec58a6-23f3-44c3-ab84-af06a7067891)
I'm having this issue when exporting an onnx model

**Comment by GavinYang5** [comment]
I am a beginner in torch exporting onnx
I tried to find a case about exporting ONNX models from Torch models for point cloud registration, but I failed
I am trying to find this error, and some online opinions suggest avoiding using torch.bool type Tensors
Because GeoTransformer uses a large number of torch.bool type tensors as indexes, and uses dict as input in forward() (isn't it unreasonable?)

**Comment by fabio-sim** [comment]
I've seen this error before (see this [related stackoverflow issue](https://stackoverflow.com/questions/70166946/pytorch-model-tracing-not-working-we-dont-have-an-op-for-atenfill)). I believe this occurs in the following lines:

https://github.com/qinzheng93/GeoTransformer/blob/e7a135af4c318ff3b8d7f6c963df094d7e4ea540/geotransformer/modules/ops/pointcloud_partition.py#L87-L88

```python
node_masks = torch.zeros(nodes.shape[0], dtype=torch.bool).cuda()  # (M,)
node_masks.index_fill_(0, point_to_node, True)
```

A possible solution is to change them to:
```python
node_masks = torch.zeros(nodes.shape[0]).cuda()  # (M,)
node_masks.index_fill_(0, point_to_node, 1)
```

**Comment by GavinYang5** [comment]
Yes, I noticed the issue you mentioned, I have made as many modifications as possible to avoid using torch.bool,But the error still exists TAT

**Comment by fabio-sim** [comment]
I see.. Then perhaps GeoTransformer is too complicated to convert to ONNX

**Comment by GavinYang5** [comment]
The model I want to export is this https://github.com/qinzheng93/GeoTransformer/blob/e7a135af4c318ff3b8d7f6c963df094d7e4ea540/experiments/geotransformer.3dmatch.stage4.gse.k3.max.oacl.stage2.sinkhorn/model.py#L19
Do you think there is any good way to gradually investigate which part of the problem is causing it? Currently, I can only receive an error message, but I don't know exactly where it was caused, so I am at a loss as to how to troubleshoot the problem

**Comment by fabio-sim** [comment]
Yes, if you're using VSCode to debug the export process, you can place breakpoints in the forward passes of modules, starting from the beginning and gradually advancing further until VSCode hits that error instead of a breakpoint. That should give you an idea of the location of the problematic op.

**Comment by GavinYang5** [comment]
Okay, thank you very much for your valuable feedback. I am ready to continue trying it out

**Comment by Dannyw911** [comment]
I encountered the similar problem.
When I exported ONNX model for this work [https://github.com/zhiyuanYU134/MIRETR](url), I got error like this: 

<img width="1326" height="401" alt="Image" src="https://github.com/user-attachments/assets/50e175c1-09e7-4c1c-b4ff-2e6fa1ad0517" />

It's because the function torch.svd it not supported for ONNX. 
I also saw your issue in [https://github.com/qinzheng93/GeoTransformer/issues/90](url)
and I checked GeoTransformer used the sample function **procrustes.py** just like **weighted_procrustes** in MIRETR.
So I think we have the same issue: some function in model is not supported for ONNX.

Have you solved this problem?
